packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityMethodChannelHandler.java

```diff
@@ -0,0 +1,163 @@
+package io.flutter.plugins.connectivity;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.ConnectivityManager;
+import android.net.Network;
+import android.net.NetworkCapabilities;
+import android.net.NetworkInfo;
+import android.net.wifi.WifiInfo;
+import android.net.wifi.WifiManager;
+import android.os.Build;
+import io.flutter.plugin.common.EventChannel;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+
+public class ConnectivityMethodChannelHandler
+    implements MethodChannel.MethodCallHandler, EventChannel.StreamHandler {
+
+  private final Context context;
+  private ConnectivityManager manager;
+  private BroadcastReceiver receiver;
+
+  public ConnectivityMethodChannelHandler(Context context) {
+    assert (context != null);
+    this.context = context;
+    this.manager =
+        (ConnectivityManager)
+            context.getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+  }
+
+  @Override
+  public void onListen(Object arguments, EventChannel.EventSink events) {
+    context.registerReceiver(receiver, new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION));
+  }
+
+  @Override
+  public void onCancel(Object arguments) {
+    context.unregisterReceiver(receiver);
+    receiver = null;
+  }
+
+  @Override
+  public void onMethodCall(MethodCall call, MethodChannel.Result result) {
```

Comment:
Would you mind adding unit tests for this class to ensure that incoming messages are correctly parsed, and outgoing messages are correctly serialized?

Reply:
I think that Dart e2e integration tests would be acceptable as an alternative to unit tests.